test.js

```diff
@@ -231,91 +194,125 @@ describe('gulp-gh-pages', function() {
-      cacheDir: path.join(__filename, 'dir')
-    })
-    .on('error', function(err) {
-      assert(err.code);
-      done();
-    })
-    .on('end', function() {
-      done(new Error('Expected an error.'));
-    })
-    .end(tmpFile);
-  });
+  it('should specify the cache directory path with `cacheDir` option',
+    function(done) {
+      ghPages({
+        remoteUrl: remoteUrl,
+        cacheDir: '__cache__',
+        push: false
+      })
+      .on('error', done)
+      .on('data', function(file) {
+        assert(file.isBuffer());
+      })
+      .on('end', function() {
+        readRemoveFile(path.join('__cache__', tmpStr), function(readErr, buf) {
+          assert.strictEqual(readErr, null);
+          assert.equal(String(buf), tmpStr);
+          done();
+        });
+      })
+      .end(tmpFile);
+    });
 
-  it('should emit an error when the repository doesn\'t exist', function(done) {
-    ghPages({remoteUrl: 'https://_/_this_/_repo_/_does_/_not_/_exist_/_.git'})
-    .on('error', function(err) {
-      assert(err);
-      done();
-    })
-    .on('end', function() {
-      done(new Error('Expected an error.'));
-    })
-    .end(tmpFile);
-  });
+  it('should emit an error when it fails to create a cache directory',
+    function(done) {
+      ghPages({
+        remoteUrl: remoteUrl,
+        cacheDir: path.join(__filename, 'dir')
+      })
+      .on('error', function(err) {
+        assert(err.code);
+        done();
+      })
+      .on('end', function() {
+        done(new Error('Expected an error.'));
+      })
+      .end(tmpFile);
+    });
 
-  it('should emit an error when the user has no permission to push the repo', function(done) {
-    ghPages({remoteUrl: 'https://' + accessToken + '@github.com/bot/move.git'})
-    .on('error', function(err) {
-      assert(/Permission to/.test(err.message));
-      done();
-    })
-    .on('end', function() {
-      done(new Error('Expected an error.'));
-    })
-    .end(tmpFile);
-  });
+  it('should emit an error when the repository doesn\'t exist',
+    function(done) {
+      ghPages({remoteUrl: 'https://_/_this_/_repo_/_does_/_not_/_exist_/_.git'})
+      .on('error', function(err) {
+        assert(err);
+        done();
+      })
+      .on('end', function() {
+        done(new Error('Expected an error.'));
+      })
+      .end(tmpFile);
+    });
 
-  it('should emit an error when the remote URL is not a git repository\'s URL', function(done) {
-    ghPages({remoteUrl: 'https://example.org/'})
-    .on('error', function(err) {
-      assert(/'https:\/\/example\.org\/' not found/.test(err.message));
-      done();
-    })
-    .on('end', function() {
-      done(new Error('Expected an error.'));
-    })
-    .end(tmpFile);
-  });
+  it('should emit an error when the user has no permission to push the repo',
+    function(done) {
+      ghPages({remoteUrl: 'git@github.com:bot/move.git'})
+      .on('error', function(err) {
+        assert(err);
```

Comment:
It was failing for me when running `mocha` locally. The error message I get is just "Command failed: /bin/sh -c git push --set-upstream origin gh-pages". Maybe something different in my environment? I don't know, but it seems too brittle to test for a specific human-readable error message in a third-party library.